### ISSUE TITLE
[DeadBeef 1.8.4] Playback pauses when reaching end of stream for multichannel (4.0) FLAC files

### ISSUE THREAD

**Issue by Drag0nFly** [DeaDBeeF-Player/deadbeef#2504]
### Steps to reproduce the problem
Play a multichannel/quadraphonic FLAC (4.0) with embedded .cue file either all the way through, or skip till the very end. 

### What's going on? Describe the problem in as much detail as possible.
Firstly, this appears to be a regression. 

I have used DeaDBeef for a number of years, including with these exact same files which now appear to have issues, and only noticed the issue on 1.8.4.

Last known-working version was 1.8.2 (compiled from source)

The file(s) all play fine until reaching the end (tested several), whereupon DeaDBeef goes into paused state. None of the other albums in my catalog have this issue (ranging from 44.1kHz files to 24-bit 96kHz, which is the same quality as the quad files; although number of channels are different.)

All the other files also have embedded .cue files, so that is not an issue.

I can provide examples of these files in order to better diagnose, if needed. (I realize quadraphonic FLACs are not that readily available.)

### Information about the software:

Deadbeef version:
1.8.4

OS: ArchLinux / Parabola Linux libre rolling

**Comment by Alexey-Yakovenko** [comment]
does it have to be a file with cuesheet? would this happen on a regular single-track 4-channel flac?

**Comment by Drag0nFly** [comment]
Mentioned the cuesheet as every file I have contains them. I'll remove it and re-test.

**Comment by Drag0nFly** [comment]
No, it's not related to the cuesheet. Stripped that from the FLAC, and the behaviour is the same.
(I first thought it perhaps was related to some incorrect timing issue in the embedded cue file)

**Comment by Alexey-Yakovenko** [comment]
Right now I'm testing with a 4 channel flac file, no cuesheet.
Created by generating a 4-channel tone using ffmpeg:
```
ffmpeg -f lavfi -i "sine=frequency=1000:duration=5" -ac 4 out.wav
```
Then converting to flac:
```
flac --channel-map=none out.wav
```

The described problem is not happening with either pulseaudio or alsa output.
In the next comment I'll report what I see when I add a cuesheet.

**Comment by Drag0nFly** [comment]
This is with ALSA, using a USB DAC (optical S/PDIF to the amp). I was thinking perhaps there are some changes wrt. handling multichannel streams over toslink, although it appears to behave the same way (i.e, 2-channels of the multichannel source being played)

**Comment by Drag0nFly** [comment]
(and it is the same USB DAC, a HA-INFO (Phiree) U2, which has been used with DeaDBeef since 2013)

**Comment by Alexey-Yakovenko** [comment]
With cuesheet: I can't get anything to play at all. Playback is stuck at position 0:00 always. So I suppose, even if the behavior is not the same -- at the very least it's related to what you're observing.

**Comment by Drag0nFly** [comment]
Interesting, as the ones I have tested with embedded cuesheet do work. But those have both types of cuesheet, i.e, the one in the metadata block #2 (VORBIS_COMMENT) and block #3 (the old, "proper" cuesheet).

But again, all those files work perfectly with DeaDBeef.

**Comment by Alexey-Yakovenko** [comment]
It turned out, something went wrong with virtualbox, and the guest ubuntu sound driver died, later bringing the whole system down :)

I restarted and tried again, and then everything worked fine. No freezing on pausing.

Btw, when the problem happens for you -- is it recoverable? Can you still play another track, or quit the player without deadlock?

**Comment by Drag0nFly** [comment]
Yes, it pauses without any error to the console, and one can just skip to the next track/album. It is pretty weird if you ask me. The affected FLAC files are also not at the end of the playlist, or something which (might) explain this.

**Comment by Alexey-Yakovenko** [comment]
For completeness, I just validated on mac, and it also works fine (with cue)

**Comment by Alexey-Yakovenko** [comment]
Changed back to Triage, since I can't reproduce, but will keep looking.. For now I'm out of ideas.
But if I may suggest -- can you try with internal dac and see if the same problem would still happen?

**Comment by Drag0nFly** [comment]
I'll try that later and report back.

**Comment by Drag0nFly** [comment]
Tried with the built-in output; it works. It only outputs through an RCA jack though, and does not change the signal type when switching from 44.1kHz and 96kHz.

Somehow I think this makes it seem even stranger, as I would expect the external DAC to have the same issue when, for example, playing a 96kHz album and switching to a 44.1kHz album -- which is not the case.

It only occurs for multichannel. Has there been some code changes to, perhaps, transcode files with more than 2 channels in order to output over S/PDIF?

I have another USB DAC as well that I can test to see if it behaves the same way, but again, it has worked 100% with this playlist before (and has played the quad FLACs probably 30 times)

**Comment by Alexey-Yakovenko** [comment]
Another suggestion with a low chance of success, but still a chance: try deadbeef 1.8.5 or 1.8.6-rc1.
There was a semi-related fix at some point, but it's hard to correlate whether it's the same case or not.

**Comment by Drag0nFly** [comment]
Tried compiling v1.8.5, but it seems to be missing some headers or a library. 

`make[2]: *** No rule to make target 'mp4p/src/mp4p.c', needed by 'mp4p/src/libmp4p_a-mp4p.o'.  Stop.`

Installed libmp4v2, which I determined was a viable candidate, but still the same make error.

**Comment by kuba160** [comment]
Did you do `git submodule update --init`?

**Comment by Drag0nFly** [comment]
I'm building via makepkg, from the 1.8.5 sources

**Comment by Alexey-Yakovenko** [comment]
A better suggestion: use prebuilt binaries

**Comment by Drag0nFly** [comment]
There are none, latest version is the one I am at, 1.8.4

**Comment by Alexey-Yakovenko** [comment]
I'm not sure where you are looking for binaries, but they are right there in the download section on deadbeef.sf.io

**Comment by Drag0nFly** [comment]
Binaries for the distro. But shouldn't the ones on https://github.com/DeaDBeeF-Player/${pkgname}/archive/${pkgver}.tar.gz
work?

If not, I can try a pre-built static binary.

**Comment by Alexey-Yakovenko** [comment]
I don't use github for publishing releases, but maybe github itself generates some files. I don't know if they work for any purposes, never tried.

**Comment by Drag0nFly** [comment]
Got a release from the sourceforge.net pages for the project. "deadbeef.sf.io" does not resolve on my end.

**Comment by Drag0nFly** [comment]
The statically-linked 1.8.5 version works(!), proceeds to the next track after playing the quadraphonic 4.0 FLAC. Surprising.

**Comment by Alexey-Yakovenko** [comment]
sorry, I should have typed https://deadbeef.sourceforge.io.
Since 1.8.5 works -- it means the bug has been fixed there.
I recommend running 1.8.6-rc1 though.. it's here:
https://sourceforge.net/projects/deadbeef/files/travis/linux/1.8/

1.8.5 has a few things that don't work, that's why 1.8.6 is going to be released almost immediately after 1.8.5
